lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart

```diff
@@ -0,0 +1,188 @@
+  });
+
+  RecommendationDetailListItemStyle copyWith({
+    TextStyle titleTextStyle,
+    TextStyle subtitleTextStyle,
+    double iconSize,
+  }) {
+    return RecommendationDetailListItemStyle(
+      titleTextStyle: titleTextStyle ?? this.titleTextStyle,
+      subtitleTextStyle: subtitleTextStyle ?? this.subtitleTextStyle,
+      iconSize: iconSize ?? this.iconSize,
+    );
+  }
+}
+
+class _DefaultStyle extends RecommendationDetailListItemStyle {
+  final TextStyle titleTextStyle = const TextStyle(
+    fontSize: 17,
+    fontWeight: FontWeight.w500,
+    color: Color(0xff4c4e6e),
+  );
+
+  final TextStyle subtitleTextStyle = const TextStyle(
+    fontSize: 15,
+    color: Color(0xff767690),
+  );
+
+  final double iconSize = 20;
+
+  const _DefaultStyle({
+    TextStyle titleTextStyle,
+    TextStyle subtitleTextStyle,
+    double iconSize,
+  });
+}
+
+const RecommendationDetailListItemStyle _defaultStyle = const _DefaultStyle();
+
+class RecommendationDetailListItem extends StatelessWidget {
+  final RecommendationDetailListItemStyle _style;
+  final RecommendationDetailListItemViewModel _viewModel;
+
+  RecommendationDetailListItem({
+    Key key,
+    RecommendationDetailListItemViewModel viewModel,
+    RecommendationDetailListItemStyle style = _defaultStyle,
+  })  : this._viewModel = viewModel,
+        this._style = style,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return ListTile(
+      leading: _buildLeading(),
+      title: _buildTitle(),
+      subtitle: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          _buildSubtitle(),
+          _buildCirclesWrap(),
+        ],
+      ),
+    );
+  }
+
+  _buildLeading() {
+    return Container(
+      padding: _Constants.leadingPadding,
+      width: _Constants.leadingContainerHeight,
+      alignment: Alignment.topRight,
+      child: Image.asset(
+        _viewModel.iconPath,
+        width: _style.iconSize,
+        height: _style.iconSize,
+      ),
+    );
+  }
+
+  _buildTitle() {
+    return Padding(
+      padding: _Constants.titlePadding,
+      child: Text(
+        _viewModel.title,
+        style: _style.titleTextStyle,
+      ),
+    );
+  }
+
+  _buildSubtitle() {
+    return Text(
+      _viewModel.subtitle,
+      style: _style.subtitleTextStyle,
+    );
+  }
+
+  _buildCirclesWrap() {
+    return Padding(
+      padding: _Constants.wrapColorsPadding,
+      child: Wrap(
+        spacing: _Constants.wrapSpacing,
+        runSpacing: _Constants.wrapRunSpacing,
+        children: _buildCircles(_viewModel.colors),
+      ),
+    );
+  }
+
+  List<Container> _buildCircles(List<Color> colors) => colors.map(
+        (color) => Container(
+              padding: _Constants.circlePadding,
+              child: CustomPaint(
+                painter: _DrawCircle(color: color),
+              ),
+            ),
+      ).toList();
+}
+
+class _DrawCircle extends CustomPainter {
```

Comment:
does it have to be done with custom painter, or could we just have square views with rounded corners half the width? be interested to see any performance differences.

Reply:
You're right there's a bit difference in FPS. With the custompainter it runs at 52-55 FPS and with simple container with a boxdecoration runs at 56-60 FPS.

Changed!

Reply:
perfect! 👍  good job